compiler/src/main/java/com/squareup/anvil/compiler/codegen/dagger/ProvidesMethodFactoryGenerator.kt

```diff
@@ -71,13 +76,27 @@ internal class ProvidesMethodFactoryGenerator : PrivateCodeGenerator() {
                 }
               }
               .forEach { function ->
-                generateFactoryClass(codeGenDir, module, clazz, function)
+                generateFactoryClassForFunction(codeGenDir, module, clazz, function)
+              }
+
+          clazz
+              .properties(includeCompanionObjects = true)
+              .asSequence()
+              .filter { property ->
+                val annotation = property.findAnnotation(daggerProvidesFqName)
+                    ?: return@filter false
+
+                // Must be '@get:Provides'.
+                annotation.useSiteTarget?.text == "get"
```

Comment:
nit: could make this filter a single line:

```
// Must be '@get:Provides'.
property.findAnnotation(daggerProvidesFqName)?.useSiteTarget?.text == "get"
```

Reply:
Oh, yes.